src/EventTimer.ts

```diff
@@ -173,7 +175,16 @@ export class EventTimer {
 				: {};
 	}
 
-	setProperty(name: 'adSlotsInline' | 'adSlotsTotal', value: number): void {
+	/**
+	 * Adds an event timer property
+	 *
+	 * @param {string} name - the property's name
+	 * @param {value} number - the property's value
+	 */
+	setProperty(
+		name: 'adSlotsInline' | 'adSlotsTotal' | 'pageHeightVH',
```

Comment:
Is there a clever typescripty way to do this? We need to limit `name` to all of the members of `EventTimerProperties` whose type is `number`

Reply:
There might be, but I'm inclined to leave it as it is here. It's quite clear, and also if you tried to pull in all the keys with values of type `number`, you'd end up with `'downlink'` as well?

Reply:
Something like (?):

```js
type ValidKeys<Type> = keyof Type;

setProperty(
	name: ValidKeys<
		Pick<
			EventTimerProperties,
			'adSlotsInline' | 'adSlotsTotal' | 'pageHeightVH'
		>
	>,
	value: number,
): void {
	this.properties[name] = value;
}
```

However I'm not sure if this is crossing that line of 'too clever and completely unreadable' .

There's a bunch of TS that makes me go cross-eyed and brain-hurty but where that line is is very subjective.